### ISSUE TITLE
Update simple perceiver to be python correct.

### ISSUE THREAD

**Issue by pv33** [ivapylibs/perceiver#1]
Most of the Matlab has been copy pasted and some python syntax modifications made.  
Code is not the same as perceiver.simple because that one was coded incorrectly relative to the API.

Recreating the hand tracker will involve designing the proper two-pass detector.
The track pointer should have been a centroid one, so that code should be done first, then applied to grayscale image as per testing/simple01.py code stub.

Complete the simple01.py code stub to flesh out the class, the implementation, and the test.

**Comment by pv33** [comment]
Combines a ``detector.inImage`` that uses thresholding via an ``improcessor.basic`` with a ``trackpointer.centroid`` to pull out the track point.  First test ``simple01_xxx`` is for a single image right now.  Just want to test functionality of API.

**Comment by pv33** [comment]
@Uio96 
I am not sure who wrote the ``simple`` perceiver, but it definitely should not be processing depth **and** color imagery. The documentation in the first 20 lines is incorrect, as is some code later on that queries color and depth data. Some other stuff is off but can be corrected later on.  I don't know if it got added in after I wrote the code stub or if it was there from the beginning but simply ignored by me. In ivaMatlibs ``markerTracker.rigidTriangle`` is a good template to follow regarding the member functions.  All of the other implementations have much more functionality coded in.

For color + depth, the expectation is that the measurement will be a structure with both data types packaged in it (depth + color). In regular python, both images would be provided at the same time.  In ROS, Shiyu and Justin recommend using a special ROS node that grabs from both topics and merges them into a joint topic. The perceive would subscribe to the joint topic.

There are possible alternatives, but those should be taken care of later.  For now, focus on a simple single image perceiver class.

**Comment by Uio96** [comment]
Yiye developed the Matlab version and I translated it to Python. The version you just saw may not be updated yet. 

I just updated that part and added the testing scripts. So the minimal functionality has been implemented as you described in the testing scripts.

**Comment by yiyeChen** [comment]
Yes I developed the ```simple``` in Matlab, and I acknowledge that the design is off since I was just trying to test something quick. It needs to be redesigned.

But I have a question about packaging multiple inputs. I am trying to develop a [```ROSWrapper```](https://github.com/ivaROS/ROSWrapper) which ideally can subscribe to any number of topics and register a corresponding callback. The core function I used is the [TimeSynchronizer](http://wiki.ros.org/message_filters), which takes in a subscribers list and a callback function *with the same number of inputs*.

I think comparing to a special node, this method is more flexible since you don't have to develop a new node for each possible topics combination. You just need to feed in the subscribers. But it will requires the callback function to accept multiple inputs directly, not packaging them. So I am not sure whether combining the inputs is a good idea or not
